server/engine/engine.py

```diff
@@ -365,6 +380,154 @@ def processLocalSubjectFolder(path: str, outputName: str = None, href: str = '')
 
         trialProcessingResults.append(trialProcessingResult)
 
+        # If ground reaction forces were provided, trim the marker and force data to the
+        # intersection of the time ranges between the two.
+        if len(trialForcePlates[itrial]) > 0 and not disableDynamics:
```

Comment:
I'm not entirely sure how to do it, but this seems like we ought to have unit tests for this much logic. I just don't trust myself to catch all the bugs in here by eye. Perhaps we could factor this out to one or more "generic utility" methods that come with unit tests, and don't need an entire pipeline in order to set up the unit tests?